### ISSUE TITLE
Deprecate `Object >> #split:`?

### ISSUE THREAD

**Issue by jordanmontt** [guillep/pharo-core#17196]
The method `Object >> #split` is understood by all objects. The method's comment says

>> Split the argument using the receiver as a separator. Optimized version for single delimiters

The question is, does it make sense to have that method in `Object`? For example, if we do

```st
SpPresenter new split: 'hello'` "'hello'"
```

It will return `'hello'` which, for me, it does not make sense.

**Comment by astares** [comment]
We could move the code (switching receiver and argument) and have a #splitUsing: in SequenceableCollection, 

```Smalltalk
'Hello' splitUsing: $l
```

We can use it in Object>>split: but deprecate this one.
```Smalltalk
split: aSequenceableCollection

  ^ aSequenceableCollection splitUsing: self
```

**Comment by jordanmontt** [comment]
Yes, we can have a `splitUsing:`. But, we will need to refactor `SequenceableCollection>>#splitOn:` to that does the same thing but inversely.

For `Object >> #split:` I'd say to remove it. Because the method is not clear. It should not be understood by all objects. The SpPresenter is a good example. Why a presenter understand `split:`

**Comment by astares** [comment]
Remove yes - but as there might be users we should deprecate it for at least one iteration. As usual.

**Comment by jordanmontt** [comment]
Yes, I meant to deprecate it to remove it in the next Pharo version
